include/bredis/Result.hpp

```diff
@@ -0,0 +1,49 @@
+//
+//
+// Copyright (c) 2017 Ivan Baidakou (basiliscos) (the dot dmol at gmail dot com)
+//
+// Distributed under the MIT Software License
+//
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include <boost/system/error_code.hpp>
+#include <boost/utility/string_ref.hpp>
+#include <boost/variant.hpp>
+
+#include "Markers.hpp"
+
+namespace bredis {
+
+template <typename DynamicBuffer> struct to_iterator {
+    using iterator_t = boost::asio::buffers_iterator<
+        typename DynamicBuffer::const_buffers_type, char>;
```

Comment:
I don't think this is a good idea at all. You are adapting a sequence of buffers into a sequence of characters, and doing so at a very low level in your library. What if the user is okay with a sequence of buffers? They will suffer performance loss. What you want is an adapter that lazily represents a sub-range of an existing **ConstBufferSequence** in an efficient way, so the caller can still see big pieces of buffers. One example is `beast::buffer_prefix`, see:
https://github.com/vinniefalco/Beast/blob/master/include/beast/core/buffer_prefix.hpp#L153

If you combine `buffer_prefix` with something like `consuming_buffers` you can efficiently adapt a buffer sequence to a sub-range without reducing the strength of your type to **ForwardRange** on `char`:
https://github.com/vinniefalco/Beast/blob/master/include/beast/core/consuming_buffers.hpp#L22